launcher/Version.cpp

```diff
@@ -74,12 +74,28 @@ bool Version::operator!=(const Version &other) const
 void Version::parse()
 {
     m_sections.clear();
-
-    // FIXME: this is bad. versions can contain a lot more separators...
-    QStringList parts = m_string.split('.');
-
-    for (const auto& part : parts)
-    {
-        m_sections.append(Section(part));
+    QString currentSection;
+    auto classChange = [] (QChar lastChar, QChar currentChar) {
+        return (( lastChar.isLetter() && currentChar.isDigit() ) || (lastChar.isDigit() && currentChar.isLetter()) );
+    };
+    for (int i = 0; i < m_string.size(); ++i) {
+        if(m_string[i].isDigit() || m_string[i].isLetter()){
```

Comment:
please run `clang-format` on here

Reply:
done in https://github.com/PrismLauncher/PrismLauncher/pull/751/commits/624a59631eea9ab834f430f2ca259cbe8c6d3c57